### ISSUE TITLE
Projects: no Validation Rule for Products in Phases and Taks

### ISSUE THREAD

**Issue by marcalwestf** [adempiere/adempiere#2210]
It is possible to select products in the _Phase_ and _Task_ tab of the _Project_ window.

In the present implementation, **any product** can be chosen, which is incorrect and can lead to problems when trying to generate for example a Sales Order. 
If in this example the product doesn't belong to a valid version of a price list or to the version selected, the sales order will fail.

Therefore, a validation rule is needed that considers
- if there is a BP defined in Project
- the Price List or Price List Version defined for the Project
- otherwise, the Price List of the Project BP
- the product must be comprised in on of the price lists/versions mentioned above.
Otherwise, no product can be chosen.

**Comment by e-Evolution** [comment]
@marcalwestf , the product can be defined while the project type from a standard phase or standard task.

So the validation should be included into the process to create a the Sales Order not when adding a phase or task , because if a producto is defined in standard phase or task not is posible validate price list

**Comment by marcalwestf** [comment]
@e-Evolution thanks for the observation. Yes, you're right: the product can be defined in Project Type and the whole Project Type structure -including products- can be copied to a project.
So, validating from the start that every product has to be in the in a Price List would not work.

Nevertheless, we have to think of and implement a way of controlling the situation. 

If we for example clutter the project with many products that are not in the Project's Price List, we can get undesired results (missing products in Order lines for example). The bigger the project the more difficult to gain control on this matter. Expecting that users somehow go to every phase and task and check it manually will end in frustration and anger.

I couldn't say off-the-sleeve what it would be, but something has to be done this regarding.

**Comment by e-Evolution** [comment]
@marcalwestf yes is simple we need adding this validation when a sales order is created from a project , so before the create the sales order head and lines , a previos validation should be the product use need are into the price list show a message ,  this products not found into the price list , the similar validation exist when entry sales order line.
